homeassistant/components/vicare/climate.py

```diff
@@ -246,22 +275,30 @@ def preset_modes(self):
         """Return the available preset mode."""
         return list(VICARE_TO_HA_PRESET_HEATING)
 
-    def set_preset_mode(self, preset_mode):
+    async def async_set_preset_mode(self, preset_mode):
         """Set new preset mode and deactivate any existing programs."""
         vicare_program = HA_TO_VICARE_PRESET_HEATING.get(preset_mode)
         if vicare_program is None:
-            _LOGGER.error(
+            raise ValueError(
                 "Cannot set invalid vicare program: %s / %s",
                 preset_mode,
                 vicare_program,
             )
-            return
 
         _LOGGER.debug("Setting preset to %s / %s", preset_mode, vicare_program)
-        self._api.deactivateProgram(self._current_program)
-        self._api.activateProgram(vicare_program)
+        await self.hass.async_add_executor_job(
+            self._async_set_program, self._current_program
+        )
 
     @property
-    def device_state_attributes(self):
+    def async_device_state_attributes(self):
         """Show Device Attributes."""
         return self._attributes
+
+    async def async_set_vicare_mode(self, mode):
```

Comment:
The service target methods can be sync.
```suggestion
    def set_vicare_mode(self, mode):
```